### ISSUE TITLE
SecItemAdd status = -34018 

### ISSUE THREAD

**Issue by maxfornacon** [mogol/flutter_secure_storage#293]
I'm trying to store and read a value in a FlutterApp running on MacOS using prerelease `5.0.0-beta.4`.

I just straight up create a storage then store a value and immediately try to read it.
Every time after the write  I get the log output `SecItemAdd status = -34018` and the read value is `null`.

I couldn't find any information regarding this status code.

```
final storage = FlutterSecureStorage();
await storage.write(key: 'key', value: 'TEST');
print('READ: ${await storage.read(key: 'key')}');
```

I added Keychain Sharing as a capability to the Runner.

flutter doctor -v output:

```
[✓] Flutter (Channel stable, 2.2.3, on macOS 11.4 20F71 darwin-arm, locale en-DE)
    • Flutter version 2.2.3 at /Users/maxfornacon/development/flutter
    • Framework revision f4abaa0735 (7 weeks ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[!] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/maxfornacon/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/macos#android-setup for more details.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 11.4 20F71 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 92.0.4515.159

! Doctor found issues in 1 category.
```

Couldn't test this on any other OS yet.

**Comment by juliansteenbakker** [comment]
Can you try again with v5.0.0-beta.5?

**Comment by maxfornacon** [comment]
Unfortunately, v5.0.0-beta.5 does't solve the problem for me. Maybe I configured something wrong. 
I'm not very experienced developing apps for MacOS yet and your documentation to set up flutter_secure_storage isn't very specific about it.

**Comment by maxisme** [comment]
I also am having the same issue on `^5.0.0-beta.5`

**Comment by maxisme** [comment]
It doesn't happen when I use:
```
flutter_secure_storage:
    git:
      url: git://github.com/openresearch/flutter_secure_storage.git
      ref: feature/macos_support
```
FYI

**Comment by maxfornacon** [comment]
Doesn't work for me because the fork doesn't support null safety.

**Comment by maxfornacon** [comment]
I just realised that I was editing the Xcode project for iOS instead of MacOS the whole time. Therefore I did add Keychain Sharing to the iOS capabilities but not for MacOS.

However, I can't confirm that it solved the issue yet because since I needed to activate code signing in Xcode I can't get the project to build successfully.

I get the following output.
```
error: Provisioning profile "MacBookProvisioningProfile" doesn't include the currently selected device "MacBook Pro" (identifier xxx). (in target 'Runner' from project 'Runner')
** BUILD FAILED **
```

But this is most likely not an issue of the package.
